mlir/lib/Target/LLVMIR/AttrKindDetail.h

```diff
@@ -59,6 +59,19 @@ getAttrKindToNameMapping() {
   return kindNamePairs;
 }
 
+static llvm::DenseMap<llvm::StringRef, llvm::Attribute::AttrKind>
+getAttrNameToKindMapping() {
+  static auto attrNameToKindMapping = []() {
+    static llvm::DenseMap<llvm::StringRef, llvm::Attribute::AttrKind>
+        nameKindMap;
+    for (auto kindNamePair : getAttrKindToNameMapping()) {
+      nameKindMap.insert({kindNamePair.second, kindNamePair.first});
+    }
+    return nameKindMap;
+  }();
```

Comment:
Are we sure we are not getting into static initialization order fiasco-like situation here? I would rather have a static map that gets lazily populated on the first call.

Reply:
> I would rather have a static map that gets lazily populated on the first call.

Isn't this what's implemented here?

Reply:
Actually the static initializer inside the function runs only on the first call to the function.
It is the idiom recommended for lazy-initialization in a thread-safe way: https://isocpp.org/wiki/faq/ctors#static-init-order-on-first-use
(I'm using this pattern frequently in MLIR actually :))

Note that the code pattern you suggested would not be thread-safe: I even sometimes have "unused" static value in a function just to run an initializer once (and benefit from the thread-safety).

Reply:
That is since C++11 you can replace:
```
void foo() {
  static llvm::once_flag initializeOnce;
  llvm::call_once(initializeOnce, []() { .... });
  ...
```

with:

That is since C++11 you can replace:
```
void foo() {
  static bool initOnce = []() { .... ; return true; }();
  ...
```

Reply:
Here is an example: https://github.com/llvm/llvm-project/blob/main/mlir/lib/CAPI/ExecutionEngine/ExecutionEngine.cpp#L26-L32